org.eclipse.scanning.points/src/org/eclipse/scanning/points/serialization/PositionBean.java

```diff
@@ -12,93 +12,43 @@
 package org.eclipse.scanning.points.serialization;
 
 import java.util.Collection;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 
 import org.eclipse.scanning.api.points.AbstractPosition;
 import org.eclipse.scanning.api.points.IPosition;
 import org.eclipse.scanning.api.points.MapPosition;
-import org.eclipse.scanning.api.points.Point;
-import org.eclipse.scanning.api.points.Scalar;
-import org.eclipse.scanning.api.points.StaticPosition;
 
-class PositionBean<T extends IPosition> {
+class PositionBean {
 
 	private Map<String, Object>  values;
 	private Map<String, Integer> indices;
 	private int stepIndex;
 	private List<Collection<String>> dimensionNames; // Dimension->Names@dimension
-	private Class<T> klass;
 
 	public PositionBean() {
-
+		// to be used for JSON deserialization only
 	}
+
 	public PositionBean(IPosition pos) {
 		this.values    = pos.getValues();
 		this.indices   = pos.getIndices();
 		this.stepIndex = pos.getStepIndex();
 		this.dimensionNames = getDimensionNames(pos);
-		@SuppressWarnings("unchecked")
-		Class<T> posKlass = (Class<T>) pos.getClass();
-		this.klass = posKlass;
 	}
 
 	private List<Collection<String>> getDimensionNames(IPosition pos) {
-		if (pos instanceof AbstractPosition) {
-			return ((AbstractPosition)pos).getDimensionNames();
-		}
-
+		if (pos instanceof AbstractPosition) return ((AbstractPosition)pos).getDimensionNames();
 		return null; // Do not have to support dimension names
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)